packages/app-bridge/src/AppBridgePlatformApp.ts

```diff
@@ -16,26 +16,86 @@ import type {
+import { ErrorMessageBus, IMessageBus, MessageBus } from './utilities/MessageBus';
+import { PlatformAppContext } from './types/PlatformAppContext';
+import { generateRandomString, notify, subscribe } from './utilities';
+import { getQueryParameters } from './utilities/queryParams';
+import { InitializationError } from './errors';
+import type { ApiMethodRegistry } from './registries';
 
 export type PlatformAppApiMethod = ApiMethodNameValidator<Pick<ApiMethodRegistry, 'getCurrentUser'>>;
 
-export type PlatformAppCommand = CommandNameValidator<Record<never, never>>;
+export type PlatformAppCommandRegistry = CommandNameValidator<{
+    openConnection: void;
+}>;
+
+export type PlatformAppCommand = CommandNameValidator<Pick<PlatformAppCommandRegistry, 'openConnection'>>;
 
 export type PlatformAppState = {
     settings: Record<string, unknown>;
 };
 
-export type PlatformAppContext = {
-    marketplaceServiceAppId: string;
+type InitializeEvent = {
+    port: MessagePort;
 };
 
 export type PlatformAppEvent = EventNameValidator<
     StateAsEventName<PlatformAppState & { '*': PlatformAppState }> &
         ContextAsEventName<PlatformAppContext & { '*': PlatformAppContext }>
 >;
 
-export interface AppBridgePlatformApp<
+export class AppBridgePlatformApp implements IAppBridgePlatformApp {
+    private messageBus: IMessageBus = new ErrorMessageBus();
+    private initialized: boolean = false;
+
+    api<ApiMethodName extends keyof PlatformAppApiMethod>(
+        apiHandler: ApiHandlerParameter<ApiMethodName, PlatformAppApiMethod>,
+    ): ApiReturn<ApiMethodName, PlatformAppApiMethod> {
+        return this.messageBus.post({
+            method: 'api',
+            parameter: apiHandler,
+        }) as ApiReturn<ApiMethodName, PlatformAppApiMethod>;
+    }
+
+    async dispatch<CommandName extends keyof PlatformAppCommand>(
+        dispatchHandler: DispatchHandlerParameter<CommandName, PlatformAppCommand>,
+    ): Promise<void> {
+        if (dispatchHandler.name === 'openConnection') {
+            const { get } = this.context();
+
+            if (get().token && !this.initialized) {
+                this.initialized = true;
+                const PUBSUB_CHECKSUM = generateRandomString();
+
+                notify(Topic.Init, PUBSUB_CHECKSUM, { token: get().token });
+                const { port } = await subscribe<InitializeEvent>(Topic.Init, PUBSUB_CHECKSUM);
+                this.messageBus = new MessageBus(port);
+                return Promise.resolve<void>(void 0);
+            } else {
+                throw new InitializationError();
+            }
+        }
+    }
+
+    context(): ContextReturn<PlatformAppContext, void>;
+    context(): unknown {
+        return {
+            get: () => getQueryParameters(window.location.href),
+        };
+    }
+
+    state(): StateReturn<PlatformAppState, void>;
+    state(): unknown {
+        return undefined;
+    }
+
+    subscribe(): EventUnsubscribeFunction {
+        return () => void 0;
```

Comment:
use the subscribe to know when `openConnection` is done, so we can remove `async` from `dispatch`
```ts
appBridge.dispatch(openConnection())
appBridge.subscribe('connectionOpened', () => { [...] })
```

Reply:
Also why do we want to have openConnection() as a method? As its just an object that is returned we can also have it as a constant.

Reply:
Regarding the subscribe() pattern. As it is also not async, how do we wait in the client on this event then? 
It might be the easiest and also more understandable for the enduser, to add an initialization method that can be called that they can awaited on. It seems to me that enduser otherwise needs to know a lot to just get the appBridge started.

Reply:
Or a dispatch. Otherwise it just needs already a utility method/sugar to make it easily usable by the enduser. Lets find an easy solution here, right?

Reply:
Worked 👍